src/org/omegat/core/statistics/FindMatches.java

```diff
@@ -233,15 +233,22 @@ public void iterate(EntryKey source, TMXEntry trans) {
                     // Not all TMX entries have a source; in that case there can be no meaningful match, so skip.
                     continue;
                 }
-                if (requiresTranslation && tmen.translation == null) {
-                    continue;
+
+                if (requiresTranslation) {
+                    if (tmen.translation == null) {
+                        continue;
+                    }
+                    String sourceLang = tmen.getPropValue(ExternalTMFactory.TMXLoader.PROP_TARGET_LANGUAGE);
+                    String targetLang = tmen.getPropValue(ExternalTMFactory.TMXLoader.PROP_SOURCE_LANGUAGE);
+                    if (sourceLang.equals(targetLang)) {
+                        continue;
+                    }
```

Comment:
I don't understand how this solves the issue. Aren't you comparing the source and target of the TM entry itself? Don't we need to compare to the *project's* source language?

Reply:
A discussion here is on assumption of  `ExternalTMFactory.TMXLoader.PROP_FOREIGN_MATCH` is `true`

When _project_ 's source language is as same as _TM_ 's source language(ex. both is "en-US"), this comparison always become `false`.

When _project_ 's source language is different from _TM_ 's source language, such as
  _project_'s source language does not have region code but _TM_ source language has. 
      (ex. project source is "en" , and  TM source is "en-US" and target is "es" )

 For example,
 - TM has entries from "en-US" to "es" ,  
 - project source language is "en". 
 - TM has an entry  "en-US" : "Hello World",  "es": "Hola Mundo"

 `TMXReader2` produce  two entries
- sourceLang = "en-US", targetLang= "en-US" sourceEntry="Hello World" targetEntry="Hello World"
- sourceLang= "en-US", targetLang="es" sourceEntry="Hello World" targetEntry="Hola Mundo" 

This output is tested on 
https://github.com/miurahr/omegat/blob/master/test/src/org/omegat/core/data/ExternalTMFactoryTest.java#L178-L180

So This comparison drop a case `sourceLang="en-US", targetLang="en-US" sourceEntry="Hello World", targetEntry="Hello World"` from displaying in the pain then solve the problem.

Reply:
> Don't we need to compare to the project's source language?

_Project_'s source language is compared with TM source language at 
https://github.com/omegat-org/omegat/blob/18db07c24b44158b79cb3d29995ba4caf68ded1b/src/org/omegat/core/data/ExternalTMFactory.java#L137-L149

Here is a place the issue is caused.

When project's source language is not exactly same as TM's source language, the issue is happened.
In addition to the fact,  `ExternalTMFactory` does not check TM's _header_ srcLang and `tuv`'s lang.

It is useful when translator want to use reverse TM , for example TM source is "EN" and target is "ES" but project source is "ES" and target "EN".

Reply:
further comment?